### ISSUE TITLE
gfx_sdl2/Linux: fullscreening randomly broken

### ISSUE THREAD

**Issue by rversteegen** [ohrrpgce/ohrrpgce#1242]
In nightlies (wasn't in hróðvitnir) using gfx_sdl2 on Linux/X11/xfce4, when switching to fullscreen sometimes the image is not stretched to cover the screen, but appears at the top-left of the screen at the same zoom level as in windowed mode, and the rest of the screen is filled with garbage (no black bars). I also saw someone complain about this in Franken. The game resolution doesn't matter (seen with both 320x200 and 300x300).

This happens:
* half the time when switching with alt-enter
* never if starting in fullscreen with `--fullscreen`
* always if starting in fullscreen because that's the remembered mode

I note that in hróðvitnir, there is momentary garbage when switching from/to fullscreen which is fixed in nightlies. There are a raft of changes to gfx_sdl2 fullscreening changes in nightlies.

**Comment by bob-the-hamster** [comment]
I am not able to reproduce this bug on Debian Linux xfce4 using the current nightlies (or any other version, I have never personally seen this bug)

I tested with all the methods of switching into fullscreen, alt+enter, command line argument, and last-stat-defaulting

My version of SDL2 is 2.0.14+dfsg2-3+deb11u1

**Comment by rversteegen** [comment]
Interesting, it seems to be specific to certain SDL2 versions, or maybe builds. Can you reproduce it when running game.sh? When I run game.sh (which uses SDL 2.0.22) it happens about 80-90% of the time. When I use my distro's SDL 2.26.0 build it never happens. Probably an SDL2 bug that was fixed, although it could be a problem with Valve's build.

**Comment by bob-the-hamster** [comment]
Aha! Yes, I can reproduce it with ./game.sh

It does indeed happen roughly 50% of the time

**Comment by bob-the-hamster** [comment]
After messing around with this for a while, I found a hacky fix that prevents the fullscreen bug from happening when I run ./game.sh

However, this hack makes it so that when I run ./ohrrpgce-game instead of ./game.sh a different, slightly worse bug happens each time I return from fullscreen to windowed mode (the whole screen gets offset upward, cutting off the top part of the screen, and leaving a black box at the bottom)

```
diff --git a/gfx_sdl2.bas b/gfx_sdl2.bas
index 7ae99b94..8121d374 100644
--- a/gfx_sdl2.bas
+++ b/gfx_sdl2.bas
@@ -1094,6 +1094,10 @@ LOCAL SUB keycombos_logic(evnt as SDL_Event)
     IF evnt.key.keysym.sym = SDLK_RETURN THEN  'alt-enter (not processed normally when using SDL)
       gfx_sdl2_setwindowed(windowedmode XOR YES)
       post_event(eventFullscreened, windowedmode = NO)
+      'Repeat these calls as a hack to fix bug #1242 when running ./game.sh
+      'but adds an even worse bug when NOT using ./game.sh *crying-emoji*
+      gfx_sdl2_setwindowed(windowedmode)
+      post_event(eventFullscreened, windowedmode = NO)
     END IF
     IF evnt.key.keysym.sym = SDLK_F4 THEN  'alt-F4
       post_terminate_signal
```

So definitely NOT a good fix, but interesting that it behaves this way

**Comment by rversteegen** [comment]
Hmm. Well no need for workarounds, we can just update the SDL builds (from steam-runtime) by running `linux/update_steam_runtime_libs.sh`, which I've just done. steam-runtime is still using SDL 2.26.5.
